src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -76,8 +78,20 @@ public static bool CanOSPerformPKeyAuth()
         {
             //PKeyAuth can only be performed on operating systems with a major OS version of 6. 
             //This corresponds to windows 7, 8, 8.1 and their server equivilents.
-#if DESKTOP
-            if (Environment.OSVersion.Version.Major == 6)
+            //Environment.OSVersion as it will return incorrect information on some operating systems
+#if NET_CORE
```

Comment:
@trwalke 
A partner recommended in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1824, the use of the [version helper APIs](https://docs.microsoft.com/en-us/windows/win32/sysinfo/version-helper-apis), but I suppose for PKAuth this is not needed ?

Reply:
correct. in the linked issue in that issue, bogdan includes a SO post w/more info.

---
In reply to: [545162569](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2312#discussion_r545162569) [](ancestors = 545162569)

Reply:
I would probably just use the registry once and cache the OS version in the app. Seems like a simpler approach.

Interestingly the [version helper APIs](https://docs.microsoft.com/en-us/windows/win32/sysinfo/version-helper-apis) are recommened for this. But then they use VerifyVersionInfo function which is [deprecated](https://docs.microsoft.com/en-us/windows/win32/sysinfo/targeting-your-application-at-windows-8-1) in Windows 10 but I guess, can still work if the app has app.manifest.